### PR TITLE
Minor utilities

### PR DESCRIPTION
**Pull request by TashiWalde**
- Define section-retraction pairs. Show that in such a pair, the section is an equivalence iff the retraction is an equivalence. (This is essentially a synonym of `is-equiv-left-factor` and `is-equiv-right-factor` with a different signature.)

- Define free path space and show that the canonical map `A -> free-paths A` is an equivalence.

- Collect three different versions of the arrow type under a common heading in `05-segal-types`.

### CODE REVIEW

File: src/simplicial-hott/05-segal-types.rzk.md [hunk 1]
```diff
@@ -216,6 +216,36 @@ Extension types are also used to define the type of commutative triangles:
...
+  := \ σ → ((σ 0₂, σ 1₂), σ)
+
+#def is-equiv-free-arrow
+  ( A : U)
+  : is-equiv (Δ¹ → A) (free-arrows A) (free-arrow A)
+  :=
+    ( ( \ ((_,_), σ) → σ
+      , \ _ → refl)
+    , ( \ ((_,_), σ) → σ
+      , \ _ → refl))
```
Comment (fredrik-bakke) [inline src/simplicial-hott/05-segal-types.rzk.md:246]: This is hard to parse like this. Generally, lambdas as subexpressions should be parenthesized, and the arguments of comma pairings, when the total expression spans multiple lines. Lastly, commas should have whitespace before and after:
```suggestion
    ( ( ( \ ((_ , _) , σ) → σ)
      , ( \ _ → refl))
    , ( ( \ ((_ , _) , σ) → σ)
      , ( \ _ → refl)))
```

File: src/simplicial-hott/05-segal-types.rzk.md [hunk 2]
```diff
@@ -216,6 +216,36 @@ Extension types are also used to define the type of commutative triangles:
...
+directly.
+
+```rzk
+#def free-arrows
+  ( A : U)
+  : U
+  :=
+    Σ ((a,b) : product A A), hom A a b
+
+#def free-arrow
```
Comment (fredrik-bakke) [inline src/simplicial-hott/05-segal-types.rzk.md:234]: This is already formalized in `discrete-types` as [fibered-arr-free-arr](https://rzk-lang.github.io/sHoTT/simplicial-hott/07-discrete.rzk/#define:fibered-arr-free-arr)

Comment (TashiWalde) [reply src/simplicial-hott/05-segal-types.rzk.md:234]: I see, I had missed the relevant part of `segal-types`, where some of it is already defined.
I will batch it all there and put it under its proper header.
I would like to also keep the one I defined, since it is uncurried. That means that one does not have to invoke `total-equiv-family-of-equiv` twice.

File: src/simplicial-hott/05-segal-types.rzk.md (hunk 1 above)
Comment (fredrik-bakke) [reply src/simplicial-hott/05-segal-types.rzk.md:246]: The parentheses are redundant. They are only redundant in the sense that `rzk` will parse the code and accept it as valid proof irrespective of whether the parentheses are there or not. However, they add uniformity to the definition, and this way the reader does not have to trust that the author formatted the proof correctly to parse it correctly. This is similar to how we parenthesize single-token arguments in multi-line function applications even though that is also "redundant".

E.g.
```rzk
my-function
  ( single-token-argument-1)
  ( some more complicated expression)
```

Comment (fredrik-bakke) [reply src/simplicial-hott/05-segal-types.rzk.md]: Another valid option for formatting this proof is
```rzk
    ( ( ( \ ((_ , _) , σ) → σ) , (\ _ → refl))
    , ( ( \ ((_ , _) , σ) → σ) , (\ _ → refl)))
```

File: src/simplicial-hott/05-segal-types.rzk.md [hunk 3]
```diff
@@ -216,6 +216,66 @@ Extension types are also used to define the type of commutative triangles:
...
+  ( A : U)
+  : (arr A) → (fibered-arr A)
+  := \ k → (k 0₂ , (k 1₂ , k))
+
+#def is-equiv-fibered-arr-free-arr
+  ( A : U)
+  : is-equiv (arr A) (fibered-arr A) (fibered-arr-free-arr A)
+  :=
+    ( ( \ (_ , (_ , f)) → f , \ _ → refl)
+    , ( \ (_ , (_ , f)) → f , \ _ → refl))
```
Comment (TashiWalde) [inline src/simplicial-hott/05-segal-types.rzk.md:249]:
```suggestion
    ( ( (\ (_ , (_ , f)) → f) , (\ _ → refl))
    , ( (\ (_ , (_ , f)) → f) , (\ _ → refl)))
```

@fredrik-bakke do I undestand correctly, that you prefer adding the brackets like this (and in the other place)?

File: src/simplicial-hott/05-segal-types.rzk.md (hunk 1 above)
Comment (TashiWalde) [reply src/simplicial-hott/05-segal-types.rzk.md]: Ok, thanks!

File: src/simplicial-hott/05-segal-types.rzk.md (hunk 3 above)
Comment (fredrik-bakke) [reply src/simplicial-hott/05-segal-types.rzk.md]: Ah, hmm, I guess here you _could_ select to omit the highlighted parentheses if you want to, as it is consistent on both sides of the comma and the total pairing expression is on a single line.

File: src/hott/03-equivalences.rzk.md [hunk 4]
```diff
@@ -533,6 +533,50 @@ The retraction associated with an equivalence is an equivalence.
...
+
+```rzk
+#variable is-sec-rec-pair : is-section-retraction-pair
+
+#def is-equiv-section-is-equiv-retraction-is-section-retraction-pair
+  ( is-equiv-r : is-equiv B A r)
+  : is-equiv A' B s
+  :=
+    is-equiv-right-factor A' B A s r
+    ( is-equiv-r) ( is-sec-rec-pair)
```
Comment (fredrik-bakke) [inline src/hott/03-equivalences.rzk.md:567]: No need for a space after the second opening parenthesis on this line.

**Review by fredrik-bakke — APPROVED** [review]
I have some guiding comments on formatting and a suggested expansion on the explanation for your notion of section-retraction pairs.

I hope my comments don't come off as nitpicky. Reviewing the code is also a chance for me to reflect on the conventions and how to communicate them more clearly.

File: src/hott/03-equivalences.rzk.md [hunk 5]
```diff
@@ -533,6 +533,50 @@ The retraction associated with an equivalence is an equivalence.
       ( is-equiv-section-is-equiv A B f is-equiv-f)
 ```
 
+## Section-retraction pairs
+
+A pair of maps `s : A' → B` and `r : B → A` is a section-retraction pair if the
+composite `A' → A` is an equivalence.
```
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/hott/03-equivalences.rzk.md:539]: Note also that I put the term that is being defined in boldface (`**section-retraction pair**`).

[the remark this answers is not in the document]
Comment (TashiWalde) [reply src/hott/03-equivalences.rzk.md:539]: I did not realize that we boldface the terms to be defined. Will do so in the future.

I also was planning on expanding this section more and give a precise comparison between this notion of section-retraction-pair and the usual notion of section & retraction. Maybe we can save your suggestion until I have done that expansion and its accompanying explanation?

File: src/hott/08-families-of-maps.rzk.md [hunk 6]
```diff
@@ -308,6 +310,45 @@ equivalence.
...
+#def free-paths
+  ( A : U)
+  : U
+  := Σ ( (x, y) : product A A) , (x = y)
+
+#def constant-free-path
+  ( A : U)
+  ( a : A)
+  : free-paths A
+  := ((a,a), refl)
```
[the remark this answers is not in the document]
Comment (TashiWalde) [reply src/hott/08-families-of-maps.rzk.md:327]: I agree with the extra spacing in general and have started doing this. I feel some exception could be warranted in cases such as this, when I want to emphasize that some pairs are "more tightly bound" than others.

File: src/hott/03-equivalences.rzk.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/hott/03-equivalences.rzk.md:539]: > I did not realize that we boldface the terms to be defined. Will do so in the future.

We don't currently, but I'm proposing we do it going forward, if you agree.

> I also was planning on expanding this section more and give a precise comparison between this notion of section-retraction-pair and the usual notion of section & retraction. Maybe we can save your suggestion until I have done that expansion and its accompanying explanation?

That's fine. Are you planning on doing that in this PR? If so, please mark it as a draft.

File: src/hott/03-equivalences.rzk.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (TashiWalde) [reply src/hott/03-equivalences.rzk.md:567]: When listing a series of arguments, I have grown to like the extra space even for those that are not at the beginning of the line. Two reasons:
1) It adds visual symmetry and makes clear that all terms are on the same footing.
2) It makes it easy to add/remove additional line breaks between the arguments if needed.

File: src/hott/03-equivalences.rzk.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (TashiWalde) [reply src/hott/03-equivalences.rzk.md:539]: No, I will make a new one. I wanted to have this one merged first, since #116 relies on it.

File: src/hott/03-equivalences.rzk.md [hunk 7]
```diff
@@ -533,6 +533,50 @@ The retraction associated with an equivalence is an equivalence.
       ( is-equiv-section-is-equiv A B f is-equiv-f)
 ```
 
+## Section-retraction pairs
+
+A pair of maps `s : A' → B` and `r : B → A` is a section-retraction pair if the
```
Comment (TashiWalde) [inline src/hott/03-equivalences.rzk.md:538]:
```suggestion
A pair of maps `s : A' → B` and `r : B → A` is a **section-retraction pair** if the
```

File: src/hott/03-equivalences.rzk.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (fredrik-bakke) [reply src/hott/03-equivalences.rzk.md:567]: Alright, although I don't agree with you yet I won't block this PR on it. If this is the way you want it, I wonder if it is better to require to have line breaks between the arguments instead.

**Review by fredrik-bakke — DISMISSED** [review]

**Review by fredrik-bakke — APPROVED** [review]
